sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -457,7 +457,8 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    */
   @scala.annotation.varargs
   def text(paths: String*): Dataset[String] = {
-    format("text").load(paths : _*).as[String](sparkSession.implicits.newStringEncoder)
+    format("text").load(paths : _*).select("value")
```

Comment:
i'm actually not sure if this is the right fix, because this is silently dropping the partition values. I was thinking just not returning anything here ... but I guess this is OK. We need to document it properly though.